packages/components/src/styles/components/form/key-value-inputs.scss

```
@@ -0,0 +1,129 @@
+//
+
+@use "../../mixins/breakpoints" as *;
+
+// notice: this is a <fieldset> element
+.hds-form-key-value-inputs {
+  display: flex;
+  flex-direction: column;
+  margin: 0;
+  padding: 0;
+  border: none;
+  row-gap: 16px;
+}
+
+.hds-form-key-value-inputs__row {
+  display: flex;
+  flex-direction: column;
+  row-gap: 8px;
+  align-items: flex-start;
+
+  @include hds-breakpoint-above("sm") {
+    display: grid;
+    flex-grow: 1;
+    grid-template-columns: var(--hds-key-value-pair-columns);
+    row-gap: 0;
+    align-items: flex-end;
+
+    &.hds-form-key-value-inputs__row--first {
+      .hds-form-key-value-inputs__row-delete-button-container,
+      .hds-form-key-value-inputs__generic-container {
+        grid-row: 2;
+        grid-column: var(--hds-key-value-pair-column-index);
+      }
+    }
+
+    &:not(.hds-form-key-value-inputs__row--first) {
+      // copied styles from sr-only class
+      .hds-form-key-value-inputs__field-header {
+        position: absolute !important;
+        width: 1px !important;
+        height: 1px !important;
+        margin: -1px !important;
+        padding: 0 !important;
+        overflow: hidden !important;
+        white-space: nowrap !important; /* 3 */
+        border: 0 !important;
+        clip: rect(1px, 1px, 1px, 1px) !important; /* 1 */
+        clip-path: inset(50%) !important; /* 2 */
+      }
+    }
+  }
+}
+
+.hds-form-key-value-inputs__footer {
+  display: flex;
+  flex-direction: column;
+  row-gap: 16px;
+  align-items: flex-start;
+}
+
+.hds-form-key-value-inputs__field {
+  display: grid;
+  grid-row: span 2;
+  grid-template-rows: subgrid;
+  width: 100%; // we want the input element to fill the parent container
+
+  .hds-form-key-value-inputs__field-header {
+    align-self: flex-end;
+  }
+
+  .hds-form-select {
+    width: 100%; // we want the select element to fill the parent container
+  }
+
+  .hds-form-key-value-inputs__field-label {
+    width: fit-content; // needed to avoid extra invisible space since the <label> is interactive
+  }
+
+  .hds-form-key-value-inputs__field-header:not(.sr-only) {
+    margin-bottom: 8px; // space between label and control
+  }
+
+  .hds-form-key-value-inputs__field-control {
+    display: flex; // to avoid extra whitespace when the content is an <input>, <select>, or <textarea> (inline elements)
+    flex-direction: column;
+    align-self: flex-start;
+    justify-self: stretch; // allows <input> and <textarea> to go full-width
+    width: 100%;
+  }
+}
+
+.hds-form-key-value-inputs__field-helper-text,
+.hds-form-key-value-inputs__helper-text {
+  &:not(:first-child) {
+    margin-top: 4px;
+  }
+
+  // special case: if there are multiple helper text lines, we want to reduce the spacing between them
+  & + .hds-form-key-value-inputs__field-helper-text,
+  & + .hds-form-key-value-inputs__helper-text {
+    margin-top: 2px;
+  }
+}
+
+.hds-form-key-value-inputs__field-error,
+.hds-form-key-value-inputs__error {
+  margin-top: 12px;
+}
```

Field error should be 8px between it and the input field (close association). 

I removed the fieldset error margin because it's not needed. The gap in the footer is providing spacing between it and the button.
```suggestion
.hds-form-key-value-inputs__field-error {
  margin-top: 8px;
}
```